index 540c54a..24c5711 100644
--- a/feeds/packages/utils/collectd/Makefile
+++ b/feeds/packages/utils/collectd/Makefile
@@ -73,7 +73,6 @@ COLLECTD_PLUGINS_DISABLED:= \
        uuid \
        varnish \
        vserver \
-       write_graphite \
        write_mongodb \
        write_redis \
        write_riemann \
@@ -135,6 +134,7 @@ COLLECTD_PLUGINS_SELECTED:= \
        users \
        vmem \
        wireless \
+       write_graphite \
        write_http \
 
 PKG_CONFIG_DEPENDS:= \
@@ -329,4 +329,5 @@ $(eval $(call BuildPlugin,uptime,uptime status input,uptime,))
 $(eval $(call BuildPlugin,users,user logged in status input,users,))
 $(eval $(call BuildPlugin,vmem,virtual memory usage input,vmem,))
 $(eval $(call BuildPlugin,wireless,wireless status input,wireless,))
+$(eval $(call BuildPlugin,write-graphite,graphite output,write_graphite,))
 $(eval $(call BuildPlugin,write-http,HTTP POST output,write_http,+PACKAGE_collectd-mod-write-http:libcurl))

